File: OnlineGroceryOrderingSystemAuthenticationJava/src/main/java/com/alpha/ogo/service/RoleService.java
package com.alpha.ogo.service;


import com.alpha.ogo.domain.Role;
import com.alpha.ogo.dto.CommonResponce;

public interface RoleService {

	CommonResponce<Role> createRole(Role role);

	CommonResponce<Role> getAllRole();

	CommonResponce<Role> getRoleById(Role role);

	CommonResponce<Role> getrolebyRoleCode(String role);

	CommonResponce<Role> deleteRole(Role role);

}

File: OnlineGroceryOrderingSystemAuthentication/src/main/kotlin/com/allion/ogoss/service/RoleService.kt
package com.allion.ogoss.service

import com.allion.ogoss.domain.Role
import com.allion.ogoss.dto.CommonResponce


interface RoleService {

    fun createRole(role: Role?): CommonResponce<Role?>?

    fun getAllRole(): CommonResponce<Role?>?

    fun getRoleById(role: Role?): CommonResponce<Role?>?

    fun getrolebyRoleCode(role: String?): CommonResponce<Role?>?

    fun deleteRole(role: Role?): CommonResponce<Role?>?

}
